de.dentrassi.pm.aspect.mvnosgi/src/de/dentrassi/pm/aspect/mvnosgi/internal/VirtualizerImpl.java

```diff
@@ -43,15 +42,41 @@ public void virtualize ( final Context context )
 
     private void processVirtualize ( final Context context ) throws Exception
     {
-        final ArtifactInformation art = context.getArtifactInformation ();
+        // The group ID will be set using the provided channel meta data or the default one.
+        String groupId = null;
+
+        // Extract some informations (e.g. group ID) from the provided channel meta data.
+        final Map<MetaKey, String> channelMetaData = context.getProvidedChannelMetaData ();
+        for ( final Entry<MetaKey, String> entry : channelMetaData.entrySet () )
```

Comment:
However I think I was wrong.

Accessing `m` arguments would be O(m log n), however m is constant, so altogether it would be still considered O(log n). Compared to O(n). So accessing by get would still be faster.

An beside that ... O(?) = headache :wink: